src/js/core/UpdateStream.js

```diff
@@ -0,0 +1,81 @@
+import { Observable } from "rxjs/Observable";
+import UserSettingsStore from "#SRC/js/stores/UserSettingsStore";
+import { request } from "@dcos/http-service";
+import Util from "#SRC/js/utils/Util";
+import { SAVED_STATE_KEY } from "#SRC/js/constants/UserSettings";
+
+const CHECK_DELAY = 24 * 60 * 60 * 1000; // once every 24 hours
+
+export const UpdateStreamType = Symbol("UpdateStreamType");
+
+function getVersionFromLocalStorage() {
+  const version = Util.findNestedPropertyInObject(
+    UserSettingsStore.getKey(SAVED_STATE_KEY),
+    "dcosUIVersion.version"
+  );
+
+  return version;
+}
+
+function setVersionInLocalStorage(version) {
+  const savedStates = UserSettingsStore.getKey(SAVED_STATE_KEY) || {};
+  savedStates.dcosUIVersion = { version };
+  UserSettingsStore.setKey(SAVED_STATE_KEY, savedStates);
+}
+
+function getContentType({ action, actionType, entity, version }) {
+  return `application/vnd.dcos.${entity}.${action}-${actionType}+json;charset=utf-8;version=${version}`;
+}
+
+function showNotification(newVersion) {
+  // TODO: show notification
+  console.log(
+    "A new version of DCOS UI is available. Install " +
+      newVersion +
+      ". Dismiss."
+  );
+}
+
+setVersionInLocalStorage("2.24.0"); // TODO: Check if a version exists
+const savedVersion = Observable.of(getVersionFromLocalStorage()); // TODO: make an actual stream
+
+const fetchedVersion = request("/package/list-versions", {
+  method: "POST",
+
+  headers: {
+    "Content-Type": getContentType({
+      action: "list-versions",
+      actionType: "request",
+      entity: "package",
+      version: "v1"
+    }),
+    Accept: getContentType({
+      action: "list-versions",
+      actionType: "response",
+      entity: "package",
+      version: "v1"
+    })
+  },
+  body: JSON.stringify({ includePackageVersions: true, packageName: "dcos-ui" })
+}).retry(4);
+
+export const compare = Observable.interval(CHECK_DELAY).switchMap(() =>
+  Observable.combineLatest(
+    savedVersion,
+    fetchedVersion,
+    (savedVersion, newVersionObject) => {
+      const newVersion = Object.keys(newVersionObject.response.results)[0];
+      if (newVersion > savedVersion) {
```

Comment:
You may need a semantic version comparator, similar to the version sorting implementation that we use for sorting services by version. Reason being, the string "2.24.0" is greater than "2.100.0", but the semantic version is greater

Reply:
Should you be comparing to the `window.DCOS_UI_VERSION` ?